src/main/java/appeng/tile/spatial/SpatialAnchorTileEntity.java

```diff
@@ -0,0 +1,375 @@
+import java.util.Iterator;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+
+import com.google.common.collect.Multiset;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityType;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.common.world.ForgeChunkManager;
+
+import appeng.api.config.Settings;
+import appeng.api.config.YesNo;
+import appeng.api.networking.GridFlags;
+import appeng.api.networking.IGridNode;
+import appeng.api.networking.events.MENetworkChannelsChanged;
+import appeng.api.networking.events.MENetworkEventSubscribe;
+import appeng.api.networking.events.MENetworkPowerStatusChange;
+import appeng.api.networking.events.statistics.MENetworkChunkEvent.MENetworkChunkAdded;
+import appeng.api.networking.events.statistics.MENetworkChunkEvent.MENetworkChunkRemoved;
+import appeng.api.networking.ticking.IGridTickable;
+import appeng.api.networking.ticking.TickRateModulation;
+import appeng.api.networking.ticking.TickingRequest;
+import appeng.api.util.AECableType;
+import appeng.api.util.AEColor;
+import appeng.api.util.AEPartLocation;
+import appeng.api.util.DimensionalCoord;
+import appeng.api.util.IConfigManager;
+import appeng.api.util.IConfigurableObject;
+import appeng.client.render.overlay.IOverlayDataSource;
+import appeng.client.render.overlay.OverlayManager;
+import appeng.me.GridAccessException;
+import appeng.services.ChunkLoadingService;
+import appeng.tile.grid.AENetworkTileEntity;
+import appeng.util.ConfigManager;
+import appeng.util.IConfigManagerHost;
+
+public class SpatialAnchorTileEntity extends AENetworkTileEntity
+        implements IGridTickable, IConfigManagerHost, IConfigurableObject, IOverlayDataSource {
+
+    private final ConfigManager manager = new ConfigManager(this);
+    private final Set<ChunkPos> chunks = new HashSet<>();
+    private int powerlessTicks = 0;
+    private boolean initialized = false;
+    private boolean displayOverlay = false;
+
+    public SpatialAnchorTileEntity(TileEntityType<?> tileEntityTypeIn) {
+        super(tileEntityTypeIn);
+        this.getProxy().setFlags(GridFlags.REQUIRE_CHANNEL);
+        this.manager.registerSetting(Settings.OVERLAY_MODE, YesNo.NO);
+    }
+
+    @Override
+    public CompoundNBT write(CompoundNBT data) {
+        super.write(data);
+        this.manager.writeToNBT(data);
+        return data;
+    }
+
+    @Override
+    public void read(BlockState blockState, CompoundNBT data) {
+        super.read(blockState, data);
+        this.manager.readFromNBT(data);
+    }
+
+    @Override
+    protected void writeToStream(PacketBuffer data) throws IOException {
+        super.writeToStream(data);
+        data.writeBoolean(displayOverlay);
+        if (this.displayOverlay) {
```

Comment:
oh god. are you not sending this in a separate packet? 👀

Reply:
I did not want to touch the packets. And I do not think we have a single te sending it's own packets. So I really do not want to open that can of worms.